src/main/java/seedu/realodex/model/person/Birthday.java

```diff
@@ -0,0 +1,94 @@
+package seedu.realodex.model.person;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Represents a remark in the Realodex
+ */
+public class Birthday {
+
+    public static final String MESSAGE_CONSTRAINTS = "Birthday should be in dd-MMM-yyyy format";
+    public static final DateFormat DATE_FORMAT = DateFormat.getDateInstance(DateFormat.MEDIUM);
+    public static final String INPUT_DATE_PATTERN = "ddMMMyyyy";
+    public final Optional<Date> birthday;
+
+    /**
+     * Constructs a {@code Remark}.
+     *
+     * @param birthday A valid birthday.
+     */
+    public Birthday(String birthday) {
+        // checkArgument(isValidBirthday(birthday));
+        SimpleDateFormat formatter = new SimpleDateFormat(INPUT_DATE_PATTERN, Locale.ENGLISH);
+        formatter.setLenient(false);
+        Optional<Date> birthdayDate;
+        try {
+            birthdayDate = Optional.of(formatter.parse(birthday));
+        } catch (ParseException e) {
+            // will only reach here with empty string, other cases have been caught in isValidBirthday
+            birthdayDate = Optional.empty();
+        }
+        this.birthday = birthdayDate;
+    }
+
+    /**
+     * Returns if a given string is a valid birthday.
+     */
+    public static boolean isValidBirthday(String birthday) {
+        if (birthday.isBlank() || birthday.equals("")) {
+            return true;
+        }
+        // todo: conduct check without using try-catch block for better code quality
```

Comment:
I think it is fine for now but if we want to specify the error then it would be meaningful to do so. 

> Specify the error
as in to return if like invalid day, invalid month invalid year or something